dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/SearchFilterToAppliedFilterConverter.java

```diff
@@ -13,18 +13,27 @@
 import org.dspace.authority.service.AuthorityValueService;
 import org.dspace.core.Context;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 /**
  * This class' purpose is to convert the SearchFilter object into a SearchResultsRest.AppliedFilter object
  */
+@Component
 public class SearchFilterToAppliedFilterConverter {
 
     @Autowired
     private AuthorityValueService authorityValueService;
 
     public SearchResultsRest.AppliedFilter convertSearchFilter(Context context, SearchFilter searchFilter) {
+
         AuthorityValue authorityValue = null;
         if (searchFilter.hasAuthorityOperator()) {
+            // FIXME this is obviously wrong as it assumes that the authorityValueService is able to retrieve the label
+            // for each Authority. Indeed, the AuthorityValueService regardless to his name is specific of the
+            // "SOLRAuthority" implementation and should not have a prominent role.
+            // Moreover, it is not possible to discover which authority is responsible for the value selected in the
+            // facet as the authority is bind at the metadata level and so a facet could contains values from multiple
+            // authorities
```

Comment:
This comment sounds like something that should be logged as a JIRA issue. We could then link to the JIRA issue in this comment (to provide more info, etc).

Reply:
I think there is a misunderstanding here. If I remember correctly, this code just looks up the authority "name" (value) in the authority core if the selected facet has an authority ID. This is done so that the DSpace 7 UI can display a meaningful value instead of a UUID.

In DSpace 5 & 6 this functionality was not present yet. This resulted in the fact that users would see a meaningless UUID in the discovery search screen. For example: http://demo.dspace.org/xmlui/discover?filtertype=author&filter_relational_operator=authority&filter=3321421d-107c-4b5b-9104-fff68ff01872

Reply:
@tomdesair thanks to clarify, nevertheless the code makes the wrong assumption that an authority always comes from the authority core that is definitively not the case or the design of the authority framework. I have created a ticket for that as suggested by Tim  https://jira.duraspace.org/browse/DS-4209